packages/fxa-auth-server/lib/routes/account.ts

```diff
@@ -1027,17 +1029,49 @@ export class AccountHandler {
 
   async accountStatusCheck(request: AuthRequest) {
     const email = (request.payload as any).email;
+    const checkDomain = !!(request.payload as any).checkDomain;
+    let invalidDomain = false;
+
+    if (checkDomain) {
+      const domain = email.split('@')[1];
+      if (!TopEmailDomains.has(domain)) {
+        let mxCheck = false,
+          ipv4Check = false;
+
+        try {
+          mxCheck = await tryResolveMx(domain);
+          ipv4Check = await tryResolveIpv4(domain);
+        } catch (error) {
+          // check against email errors
```

Comment:
If we error, we silently ignore it?

Reply:
I added some logging around it. The behavior as it currently exists in the content server is that if any error happens during this check, we let the user proceed without checking if the domain is valid. Let me know if the logging I added is sufficient.